### PR TITLE
07 post api art comm

### PR DESCRIPTION
**Pull request by lanc2112b**
Implement post comment api endpoint ticket 07.

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -121,9 +122,46 @@ describe("GET Endpoints", () => {
...
+    const newComment = {
+      username: 'rogersop',
+      body: 'Some new comment for article with id: 3',
+    };
+
+    it('Adds a new comment to article, if article exists', () => {
+      return request(app)
+        .post("/api/articles/3/comments")
+        .send(newComment)
+        .expect(200)
```
Comment (yalicee) [inline __tests__/app.test.js]: expect 201 not 200

File: controllers/commentController.js [hunk 2]
```diff
@@ -19,3 +23,29 @@ exports.getArtCommentsById = (request, response, next) => {
...
+  //const articlePromise = selectArticleById(article_id);
+
+  selectArticleById(article_id)
+    .then((result) => {
+      // if we have a result here, don't care, as it hasn't already been rejected
+      // article must exist, so:
+      return insertCommentByArtId(article_id, username, body); //so not nesting then, sure there's a better way?
+    }) // hmm more comments than code
+    .then((result) => {
+      response.status(200).send({ comment: result[0] }); // respond with the newly inserted row.
```
Comment (yalicee) [inline controllers/commentController.js]: status 201 instead of 200

**Review by yalicee — CHANGES_REQUESTED** [review]
Please consider error handling for your post request ticket. there will be multiple 400 and 404s to test for

File: __tests__/app.test.js [hunk 3]
```diff
@@ -121,9 +122,46 @@ describe("GET Endpoints", () => {
...
+            author: expect.any(String),
+            body: expect.any(String),
+          });
+        });
+    });
+  });
+
+});
+
+
```
Comment (emilyb93) [inline __tests__/app.test.js:145]: This test is a great test for the happy path but we are giving the user lots of input here and so we need to expect that the user can mess it up in many ways.
Think about the following issues and how you would handle them;
- client posts a comment to an article that doesnt exist
- client posts a comment to an article id that is not a number
- client does a post requests to a valid article but forgets to send the comment
- client posts a comment but the username doesn't exist in your database

Write the tests for these and see if you can figure out how to handle them :)

File: __tests__/app.test.js [hunk 4]
```diff
@@ -121,9 +122,46 @@ describe("GET Endpoints", () => {
...
+          const { comment } = response.body;
+          expect(comment).toBeInstanceOf(Object); //Array of comment objects
+          expect(comment.article_id).toBe(3);
+          expect(comment).toMatchObject({
+            comment_id: expect.any(Number),
+            votes: expect.any(Number),
+            article_id: expect.any(Number),
+            created_at: expect.any(String),
+            author: expect.any(String),
+            body: expect.any(String),
```
Comment (emilyb93) [inline __tests__/app.test.js]: I disagree with some of this, if we are posting a comment, we know EXACTLY the details of some of this. We know that the author and body is what we sent, we know what article id it was posted to. The votes will be set by default to 0. We could even test the comment_id if we know how many comments are in the test data.

**Review by emilyb93 — CHANGES_REQUESTED** [review]
Good code but we need more tests to prove it

File: __tests__/app.test.js [hunk 5]
```diff
@@ -121,9 +122,46 @@ describe("GET Endpoints", () => {
...
+          });
+        });
+    });
```
Comment (yalicee) [inline __tests__/app.test.js:151]: the comment you post will be the comment you get back. have a think about what you can expect

**Review by yalicee — CHANGES_REQUESTED** [review]

File: __tests__/app.test.js [hunk 6]
```diff
@@ -144,4 +182,100 @@ describe("Error handling tests", () => {
       });
   });
 
+  /* Ok Beth's bad url is breaking everything, or at least my handling of it is! */
```
Comment (yalicee) [inline __tests__/app.test.js]: remove this comment

File: controllers/commentController.js [hunk 7]
```diff
@@ -19,3 +24,28 @@ exports.getArtCommentsById = (request, response, next) => {
...
+      // check for author next
+      return selectUsernameByName(username); // return or reject in model
+    }) 
+    .then((result) => { 
+
+      return insertCommentByArtId(article_id, username, body); // return or reject in model
+    })
+    .then((result) => {
+      
+      response.status(201).send({ comment: result[0] }); // respond with the newly inserted row.
```
Comment (yalicee) [inline controllers/commentController.js]: For better code, result[0] could just be result and you would make sure to send the correct data back from your model

File: models/commentModel.js [hunk 8]
```diff
@@ -16,3 +17,43 @@ exports.selectCommentsByArticleId = (id) => {
       return result.rows;
     });
 };
+
+exports.insertCommentByArtId = (id, author, body) => {
+  //console.log(body, "No Boddddddddy");
+  //console.log(author, "No Boddddddddy");
+  //console.log(id, "No Boddddddddy");
+  if (!body) {
```
Comment (yalicee) [inline models/commentModel.js]: remove log comments or any unnecessary comments

**Review by yalicee — APPROVED** [review]
Looks good :)

**Pull request merged**